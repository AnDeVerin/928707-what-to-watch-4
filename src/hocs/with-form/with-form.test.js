import React from 'react';
import renderer from 'react-test-renderer';

import withForm from './with-form.js';

const MockComponent = (props) => <div {...props} />;

const MockComponentWrapped = withForm(MockComponent);

it(`withForm is rendered correctly`, () => {
  const component = renderer
    .create(
      <MockComponentWrapped
        match={{ params: { id: 1 } }}
        location={{}}
        getMovie={jest.fn()}
        onSubmit={jest.fn()}
      />
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});
