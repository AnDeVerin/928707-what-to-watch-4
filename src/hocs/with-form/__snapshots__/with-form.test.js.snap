// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withForm is rendered correctly 1`] = `
<div
  comment=""
  getMovie={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isDisabled={false}
  isFormValid={false}
  location={Object {}}
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
  onRatingChange={[Function]}
  onSubmit={[Function]}
  onTextInput={[Function]}
  rating=""
/>
`;
